src/main/scala/redis/api/Streams.scala

```diff
@@ -0,0 +1,126 @@
+}
+
+private [redis] object Xrange {
+  def buildArgs[K](key: K, id1: RequestStreamId, id2: RequestStreamId, count: Option[Long])(implicit serializerK: ByteStringSerializer[K]): Seq[ByteString] = {
+    val builder = Seq.newBuilder[ByteString]
+    builder += serializerK.serialize(key)
+    builder += id1.serialize
+    builder += id2.serialize
+    if (count.isDefined) {
+      builder += ByteString("COUNT")
+      builder += ByteString(count.get.toString)
+    }
+    builder.result()
+  }
+}
+
+case class Xrevrange[K, F, V](key: K, end: RequestStreamId, start: RequestStreamId, count: Option[Long] = None)(implicit serializerK: ByteStringSerializer[K], deserializerF: ByteStringDeserializer[F], deserializerV: ByteStringDeserializer[V])
+  extends SimpleClusterKey[K] with RedisCommandMultiBulk[Seq[StreamEntry[F, V]]] {
+  val isMasterOnly = false
+  val encodedRequest: ByteString = encode("XREVRANGE", Xrange.buildArgs(key, end, start, count))
+  def decodeReply(reply: MultiBulk): Seq[StreamEntry[F, V]] = StreamEntryDecoder.toSeqEntry(reply)
+}
+
+case class Xtrim[K](key: K, trimStrategy: TrimStrategy)(implicit serializerK: ByteStringSerializer[K])
+  extends SimpleClusterKey[K] with RedisCommandIntegerLong {
+  val isMasterOnly = true
+  val encodedRequest: ByteString = encode("XTRIM", Seq(keyAsString) ++ trimStrategy.toByteString)
+}
+
+case class Xread[K, F, V](streams: Seq[(K, RequestStreamId)], count: Option[Long])(implicit serializerK: ByteStringSerializer[K], deserializerK: ByteStringDeserializer[K], deserializerF: ByteStringDeserializer[F], deserializerV: ByteStringDeserializer[V])
+  extends RedisCommandMultiBulk[Option[Seq[(K, Seq[StreamEntry[F, V]])]]] {
+  val keys = streams.map(_._1)
+  val isMasterOnly = false
+  val encodedRequest: ByteString = {
+    val builder = Seq.newBuilder[ByteString]
+
+    if (count.isDefined) {
+      builder += ByteString("COUNT")
+      builder += ByteString(count.get.toString)
+    }
+
+    builder += ByteString("STREAMS")
+    builder ++= streams.map(p => serializerK.serialize(p._1))
+    builder ++= streams.map(p => p._2.serialize)
+
+    encode("XREAD", builder.result())
+  }
+
+  def decodeReply(r: MultiBulk): Option[Seq[(K, Seq[StreamEntry[F, V]])]] =
+    StreamEntryDecoder.toOptionSeqStringSeqEntry(r)(deserializerK, deserializerF, deserializerV)
+}
+
+private [redis] object StreamEntryDecoder {
+  def toEntry[F, V](mb: MultiBulk)(implicit deserializerF: ByteStringDeserializer[F], deserializerV: ByteStringDeserializer[V]): StreamEntry[F, V] = {
+    val r = mb.responses.get
+    val id = StreamId.deserialize(r(0).toByteString)
+    val fields = r(1).asInstanceOf[MultiBulk].responses.get.map(_.toByteString).grouped(2).map(p => (deserializerF.deserialize(p(0)), deserializerV.deserialize(p(1)))).toSeq
+    StreamEntry(id, fields)
+  }
+
+  def toSeqEntry[F, V](mb: MultiBulk)(implicit deserializerF: ByteStringDeserializer[F], deserializerV: ByteStringDeserializer[V]): Seq[StreamEntry[F, V]] = {
```

Comment:
relatively similar to `MultiBulkConverter.toSeqByteString`

Reply:
While the method signature is similar, the underlying reply from Redis is significantly different.

The Stream commands use nested arrays in their replies to a much greater extent than commands for the other data structures. For example, `XRANGE` returns an array of two-element arrays, where the first element is the stream id and the second element is an array of field-value pairs. Borrowing JSON notation, this would be something like:

```javascript
[ 
   [ ID1, [ FIELD1, VALUE1, FIELD2, VALUE2, ... ] ], 
   [ ID2, [ FIELD1, VALUE1, FIELD2, VALUE2, ... ] ],
   ...
]
```
where `ID`, `FIELD` and `VALUE` are all Bulk strings and `[]` indicates a MultiBulk array.

The `XREAD` reply is even more nested, with an array containing two element arrays where the first element is the stream key and the second element is a sequence of entries, similar to the  `XRANGE` reply.

Because the replies are structured in such specific ways for Stream comamnds, I opted to put the decoding logic in a separate object (StreamEntryDecoder) rather than adding it to MultiBulkConverter.

I opted to use unsafe operations (casting to MultiBulk, accessing elements by index) for two reasons. First, I think it's best for decoding to break immediately and loudly if we get a reply that doesn't match our understanding/implementation of the Redis spec. Second, I don't see good fallback options, aside from silently dropping parts of the response entirely. Throwing an exception seems like the best response. It seems like there is precedent here in the use of head/tail in `MultiBulkConverter.seqtoMapString` and `MultiBulkConverter.toOptionStringByteString`, which will also throw exceptions if the array is too small.

How would you feel about introducing a new exception specifically for decode errors? This feels cleaner than emitting low-level exceptions like `ClassCastException`, `NoSuchElementException`, etc., and would allow callers to implement custom decode error handling if desired. I thought about reusing ReplyErrorException for this, but that seems best reserved for error messages from the Redis server itself. Perhaps `ReplyDecodeException`?